arrow-core-data/src/main/kotlin/arrow/core/Option.kt

```diff
@@ -495,47 +712,254 @@ sealed class Option<out A> : OptionOf<A> {
    *
    * @param predicate the predicate used for testing.
    */
-  fun filterNot(predicate: Predicate<A>): Option<A> =
+  inline fun filterNot(predicate: (A) -> Boolean): Option<A> =
     flatMap { a -> if (!predicate(a)) Some(a) else None }
 
   /**
    * Returns true if this option is nonempty '''and''' the predicate
    * $p returns true when applied to this $option's value.
    * Otherwise, returns false.
    *
+   * Example:
+   * ```
+   * Some(12).exists { it > 10 } // Result: true
+   * Some(7).exists { it > 10 }  // Result: false
+   *
+   * val none: Option<Int> = None
+   * none.exists { it > 10 }      // Result: false
+   * ```
+   *
    * @param predicate the predicate to test
    */
-  fun exists(predicate: Predicate<A>): Boolean = fold({ false }, { a -> predicate(a) })
+  inline fun exists(predicate: (A) -> Boolean): Boolean = fold({ false }, predicate)
+
+  inline fun filterA(predicate: (A) -> Iterable<Boolean>): Iterable<Option<A>> =
```

Comment:
`filterA`? Where does this name come from 🤔?

Reply:
I guess `A` comes from `Applicative` - [see here](fadb9a31ffb0a4b8f9bc88c674cefae2680a3b415cf7c5d2900aa7cd9f74afe0L77). I will rename it to `filterIterable`

Reply:
Yes, `filterIterable` would be great 👌